src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -615,6 +616,15 @@ fromChunks cs FinalChunk{..} = CompactIndex {
   Serialisation
 -------------------------------------------------------------------------------}
 
+-- | By writing out the version in host endianness, we also indicate endianness.
+-- During deserialisation, we would discover an endianness mismatch.
+indexVersion :: Word32
+indexVersion = 1
+
+-- | 32 bits, to be used before writing any other parts of the serialised file!
+versionBuilder :: BB.Builder
+versionBuilder = BB.word32Host indexVersion
```

Comment:
This doesn't appear to be used anywhere (other than the tests). I'm confused.

I'm guessing that the idea is that this `versionBuilder`, `chunkBuilder` and `finalChunkBuilder` are supposed to be used by the code that implements the incremental writing out of the index. But doesn't that code already exist, so it ought to be updated to us this?

And if that's the API for the serialisation of the index, perhaps a more general name, like "header" rather than "version" since outside of this module we don't care that it's the version, just that it has to go before the bulk of the index data.

Reply:
Oh is this because it doesn't depend on #111 and that's where it would be used?

Reply:
Yes, `chunkBuilder` is not used outside the tests either, at this point. I'm going to intregrate them in #111.